povm_toolbox/library/mutually_unbiased_bases_povm.py

```diff
@@ -0,0 +1,126 @@
+                to indicate the three Euler angles to rotate the locally-biased classical shadows
+                measurement in the Bloch sphere representation. If 2D, it will have a new set of
+                angles for each qubit. The sequence of intrinsic rotations associated with the Euler
+                angles is z-y'-z''.
+            measurement_twirl : option to randomly twirl the measurements. For each single-qubit
+                projective measurement, random twirling is equivalent to randomly flipping the
+                measurement. This is equivalent to randomly taking the opposite Bloch vector in
+                the Bloch sphere representation.
+            measurement_layout: list of indices specifying on which qubits the POVM
+                acts. If None, two cases can be distinguished: 1) if a circuit supplied
+                to the :meth:`.compose_circuits` has been transpiled, its final
+                transpile layout will be used as default value, 2) otherwise, a
+                simple one-to-one layout ``list(range(n_qubits))`` is used.
+            shot_repetitions: number of times the measurement is repeated for each
+                sampled PVM. More precisely, a new PVM is sampled for all ``shots``
+                (i.e. the number of times as specified by the user via the ``shots``
+                argument of the method :meth:`.POVMSampler.run`). Then, the parameter
+                ``shot_repetitions`` states how many times we repeat the measurement
+                for each sampled PVM (default is 1). Therefore, the effective total
+                number of measurement shots is ``shots`` multiplied by ``shot_repetitions``.
+            seed_rng: optional seed to fix the :class:`numpy.random.Generator` used to sample PVMs.
+                The Z-,X-,Y-measurements are sampled according to the probability distribution(s)
+                specified by ``bias``. The user can also directly provide a random generator. If
+                None, a random seed will be used.
+        """
+        if bias.shape[-1] != 3:
+            raise ValueError
+
+        if angles.shape == (3,):
+            processed_angles = self._process_angles(*angles)
+        elif angles.shape == (n_qubit, 3):
+            processed_angles = np.zeros((n_qubit, 6))
+            for i, angles_qubit_i in enumerate(angles):
+                processed_angles[i] = self._process_angles(*angles_qubit_i)
+        else:
+            raise ValueError
+
+        super().__init__(
+            n_qubit=n_qubit,
+            bias=bias,
+            angles=processed_angles,
+            measurement_twirl=measurement_twirl,
+            measurement_layout=measurement_layout,
+            shot_repetitions=shot_repetitions,
+            seed_rng=seed_rng,
+        )
+
+    @staticmethod
+    def _process_angles(theta: float, phi: float, lam: float) -> np.ndarray:
```

Comment:
Is $\lambda$ the conventional name for this angle?

Just asking because I would prefer a non-abbreviate argument name here, but in this case it obviously clashes with the `lambda` Python keyword.

Reply:
Related to the comment below (I followed Qiskit convention). Should I keep this name ?

Reply:
Yes let's stick to this naming if Qiskit already uses it :+1: